include/platform/toolchain.h

```diff
@@ -0,0 +1,158 @@
+/*
+ *  Copyright (c) 2016, Nest Labs, Inc.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @defgroup toolchain Toolchain
+ * @ingroup platform
+ *
+ * @brief
+ *   This module defines a toolchain abstraction layer through macros.
+ *
+ * @{
+ *
+ */
+
+#ifndef TOOLCHAIN_H_
+#define TOOLCHAIN_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+/**
+ * @def OT_TOOL_PACKED_BEGIN
+ *
+ * Indicate to the compiler that a class or struct should be byte packed.
+ *
+ * Usage:
+ *
+ *    @code
+ *
+ *    typedef
+ *    OT_TOOL_PACKED_BEGIN
+ *    struct {
+ *       char field1;
+ *       union {
+ *          char field2;
+ *          long field2;
+ *       } OT_TOOL_PACKED_FIELD;
+ *    } OT_TOOL_PACKED_END packed_struct_t;
+ *
+ *    @endcode
```

Comment:
Should this usage also be included in `OT_TOOL_PACKED_FIELD` and `OT_TOOL_PACKED_END`.